lib/bitcoin/SidetreeTransactionParser.ts

```diff
@@ -119,14 +122,27 @@ export default class SidetreeTransactionParser {
   }
 
   private async fetchOutput (transactionId: string, outputIndexToFetch: number): Promise<BitcoinOutputModel | undefined> {
-    try {
-      const transaction = await this.bitcoinClient.getRawTransaction(transactionId);
-
-      return transaction.outputs[outputIndexToFetch];
-    } catch (e) {
-      Logger.warn(`Error while trying to get outputIdx: ${outputIndexToFetch} from transaction: ${transactionId}. Error: ${SidetreeError.stringify(e)}`);
-      return undefined;
-    }
+    let shouldRetry = false;
+    let output;
+    do {
+      try {
+        const transaction = await this.bitcoinClient.getRawTransaction(transactionId);
+        output = transaction.outputs[outputIndexToFetch];
+        shouldRetry = false;
+      } catch (e) {
+        if (e instanceof FetchError && e.type === 'request-timeout') {
+          // timeout should retry until success.
+          shouldRetry = true;
+          EventEmitter.emit(EventCode.BitcoinTransactionParserTransactionFetchFailure);
+          Logger.warn(`Transaction fetch timed out for transaction id: ${transactionId}`);
+        } else {
+          // other errors mean the transaction is bad
+          shouldRetry = false;

```

Comment:
Let's discuss, not sure this is the optimal thing to do. We can never be sure that "every other error condition" means the transaction is definitely "bad" right?

So I still feel throwing is still the way to go.